lib/bolt/core/src/tasks/test.rs

```diff
@@ -519,6 +492,171 @@ fn print_results(test_results: &[TestResult]) {
 	}
 }
 
+async fn cleanup_jobs(ctx: &ProjectContext, no_purge: bool) -> Result<()> {
+	eprintln!();
+	rivet_term::status::progress("Cleaning up jobs", "");
+
+	let purge = if no_purge { "" } else { "-purge" };
+	let cleanup_cmd = formatdoc!(
+		r#"
+		nomad job status |
+			grep -v -e "ID" -e "No running jobs" |
+			cut -f1 -d ' ' |
+			xargs -I {{}} nomad job stop {purge} -detach {{}}
+		"#
+	);
+
+	let mut cmd = Command::new("kubectl");
+	cmd.args(&[
+		"exec",
+		"service/nomad-server",
+		"-n",
+		"nomad",
+		"--container",
+		"nomad-instance",
+		"--",
+		"sh",
+		"-c",
+		&cleanup_cmd,
+	]);
+	cmd.env("KUBECONFIG", ctx.gen_kubeconfig_path());
+
+	let status = cmd.status().await?;
+	ensure!(status.success());
+
+	Ok(())
+}
+
+#[derive(Deserialize)]
+struct ApiErrorResponse {
+	errors: Vec<ApiError>,
+}
+
+impl fmt::Display for ApiErrorResponse {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		for error in &self.errors {
+			if let Some(field) = &error.field {
+				write!(f, "{:?}: ", field)?;
+			}
+
+			writeln!(f, "{}", error.reason)?;
+		}
+
+		std::result::Result::Ok(())
+	}
+}
+
+#[derive(Deserialize)]
+struct ApiError {
+	field: Option<String>,
+	reason: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct TaggedObjectsListResponse {
+	data: Vec<TaggedObject>,
+}
+
+#[derive(Debug, Deserialize)]
+struct TaggedObject {
+	#[serde(rename = "type")]
+	_type: String,
+	data: Data,
+}
+
+#[derive(Debug, Deserialize)]
+struct Data {
+	id: u64,
+}
+
+// TODO: This only deletes linodes and firewalls, the ssh key still remains
+async fn cleanup_servers(ctx: &ProjectContext) -> Result<()> {
+	eprintln!();
+	rivet_term::status::progress("Cleaning up servers", "");
+
+	// Create client
+	let api_token = ctx.read_secret(&["linode", "token"]).await?;
+	let auth = format!("Bearer {}", api_token,);
+	let mut headers = header::HeaderMap::new();
+	headers.insert(header::AUTHORIZATION, header::HeaderValue::from_str(&auth)?);
+	let client = reqwest::Client::builder()
+		.default_headers(headers)
+		.build()?;
+
+	// Fetch all objects with the tag "test"
+	let test_tag = "test";
```

Comment:
is this going to be a problem with multiple tests from different clusters? shoudl we tag the test with the cluster id (like the global cluter id in the ns config, not the servers cluster id)?

Reply:
No, it deletes all servers with the "test" tag. Clusters have nothing to do with with server cleanup